asapdiscovery-data/asapdiscovery/data/tests/test_ligand_schema.py

```diff
@@ -28,6 +28,12 @@ def test_ligand_from_smiles(smiles):
     assert lig.smiles == smiles
 
 
+def test_empty_ligand():
+    lig = Ligand.from_smiles("", compound_name="test_empty")
+    assert lig.data is None
+    assert lig.to_oemol() is None
```

Comment:
Am I wrong in thinking that the `Ligand` instance shouldn't be able to return a `None` for this method? This is something that we commonly want to avoid, because this can easily lead to the infamously common "`NoneType` is not xxxxxx" error or similar further along the pipeline, which we want to avoid. We want to fail "early".

Reply:
Or in other words, this looks like working around a deeper problem in the underlying objects, namely `Ligand` in this case.

Reply:
Agreed, what do you think we should be returning instead? or set a validation error for Ligand if it is empty?

Reply:
also I figured out where the change is from pre- #692 that causes the break in #990

Reply:
yes, in my opinion having it to fail when it's `None` would be better, but that would need to be tested since the behavior is already there and some tools further down the pipeline might already be using this. Now, to be fair, according to the documentation it is not supposed to return `None`, and that's the contract we are willing to sign. So, if we are following the documentation (and type annotations) this should be "fine".